pkg/utils/util.go

```diff
@@ -137,10 +137,10 @@ func GetServiceType(runAsController, runControllerService, runNodeService bool)
 	if serviceType == 0 {
 		klog.Warning("no service type activated, this configuration may not be useful")
 	}
-	if serviceType&Controller == 0 {
+	if serviceType&Controller != 0 {
```

Comment:
Just print the serviceType; the log isn't valuable enough for two conditions.

Reply:
serviceType is a number. We need to implement String() for it to get meaningful output. And we still need branch there, maybe more complex, since we can enable controller and node service at the same time.

Constant log message is also friendly when searching for the message in code.

Reply:
all right, I thought it was simplicity, the possibilities are solely 1, 2, and 3.